schema/shared/src/main/scala/zio/blocks/schema/SchemaAspect.scala

```diff
@@ -0,0 +1,24 @@
+package zio.blocks.schema
+
+trait SchemaAspect[-Min, +Max] {
+
+  def apply[A](reflect: Reflect.Bound[A]): Reflect.Bound[A]
```

Comment:
This `A` should be constrained to be `<: Min` and `>: Max`. This means some schema aspects can be applied to only some types.

Such "constrained" schema aspects can only be applied to a single node (type safely), without recursive application.

@sviezypan Does this make sense?

Reply:
it makes sense. However can you come up with more examples of aspects that would use Min & Max? Right now I have only `identity` `doc` and `examples` aspect and none of those are using these type params really. I would like to create some more complex aspects and add some test for those.

Reply:
`examples` should use the type parameter because it can only apply to a single node (the node where the optic points to). In fact, you should not be able to write `examples` in a type-safe way without specifying `Min = A` and `Max = A`.